src/rust/ide/lib/enso-protocol/src/language_server/types.rs

```diff
@@ -19,7 +19,7 @@ pub type Event = json_rpc::handler::Event<Notification>;
 
 /// A path is a representation of a path relative to a specified content root.
 // FIXME [mwu] Consider rename to something like `FilePath`, see https://github.com/luna/enso/issues/708
-#[derive(Clone,Debug,Serialize,Deserialize,Hash,PartialEq,Eq)]
+#[derive(Clone,Debug,Default,Serialize,Deserialize,Hash,PartialEq,Eq)]
```

Comment:
I'm not convinced to the @mwu arguments - Default literally means "I don't care what the exact value is" - if someone passes such value to the functions, he should not be surprised that they fails. Moreover I think the "Valid rootId" cannot be an invariant for such structure, because I believe the roots could be attached/detached from project without our immediate knowledge.

Reply:
@farmaazon IMO Not caring about exact value is fine as long as the values remain sensible, i.e. the type does not have strong semantics.
Empty string is fine as a default -- perfectly valid string. But using an empty string for path is not nice (as you'd written in your next comment  inherently invalid) — you should never need such value. If you do, there's something wrong elsewhere.  
If you really, really need such value -- construct it by yourself, but the lack of `Default` instance serves a purpose. Reminds you that these fields really should be filled with something sensible.

Reply:
I second what @mwu-tow told. We should always think of default this way. Default doesn't matter, but allowing default create values that are clearly broken is bad.